sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/crypto/_models.py

```diff
@@ -2,11 +2,363 @@
+        format: PublicFormat,
+    ) -> NoReturn:
+        """Not implemented."""
+        raise NotImplementedError()
+
+    def verify(
+        self,
+        signature: bytes,
+        data: bytes,
+        padding: AsymmetricPadding,
+        algorithm: Union[Prehashed, HashAlgorithm],
+    ) -> None:
+        """Verifies the signature of the data.
+
+        :param bytes signature: The signature to sign, as bytes.
+        :param bytes data: The message string that was signed., as bytes.
+        :param padding: The padding to use. Supported paddings are `PKCS1v15` and `PSS`. For `PSS`, the only supported
+            mask generation function is `MGF1`. See https://learn.microsoft.com/azure/key-vault/keys/about-keys-details
+            for details.
+        :type padding: AsymmetricPadding
+        :param algorithm: The algorithm to sign with. Only `HashAlgorithm`s are supported -- specifically, `SHA256`,
+            `SHA384`, and `SHA512`.
+        :type algorithm: Prehashed or HashAlgorithm
+
+        :raises InvalidSignature: If the signature does not validate.
+        """
+        if isinstance(algorithm, Prehashed):
+            raise ValueError("`Prehashed` algorithms are unsupported. Please provide a `HashAlgorithm` instead.")
+        mapped_algorithm = get_signature_algorithm(padding, algorithm)
+        digest = Hash(algorithm)
+        digest.update(data)
+        result = self._client.verify(mapped_algorithm, digest.finalize(), signature)
+        if not result.is_valid:
+            raise InvalidSignature(f"The provided signature '{signature!r}' is invalid.")
+
+    def recover_data_from_signature(  # pylint:disable=docstring-missing-param,docstring-missing-return,docstring-missing-rtype
+        self,
+        signature: bytes,
+        padding: AsymmetricPadding,
+        algorithm: Optional[HashAlgorithm],
+    ) -> NoReturn:
+        """Not implemented."""
+        raise NotImplementedError()
+
+    def __eq__(self, other: object) -> bool:
+        """Checks equality.
+
+        :param object other: Another object to compare with this instance. Currently, only comparisons with
+            `KeyVaultRSAPrivateKey` or `JsonWebKey` instances are supported.
+
+        :returns: True if the objects are equal; False otherwise.
+        :rtype: bool
+        """
+        if isinstance(other, KeyVaultRSAPublicKey):
+            return all(getattr(self._key, field) == getattr(other._key, field) for field in self._key._FIELDS)
+        if isinstance(other, JsonWebKey):
+            return all(getattr(self._key, field) == getattr(other, field) for field in self._key._FIELDS)
+        return False
+
+    def verifier(  # pylint:disable=docstring-missing-param,docstring-missing-return,docstring-missing-rtype
+        self, signature: bytes, padding: AsymmetricPadding, algorithm: HashAlgorithm
+    ) -> NoReturn:
+        """Not implemented. This method was deprecated in `cryptography` 2.0 and removed in 37.0.0."""
```

Comment:
Is the reason we have this because we support `cryptography` < v37?

Reply:
Yes; we support 2.1.4 and above: https://github.com/Azure/azure-sdk-for-python/blob/232d010e890ba3a94cdd4fc0d97a7107e9eb1e90/sdk/keyvault/azure-keyvault-keys/setup.py#L73
We might want to evaluate our version dependency, but we were blocked from using versions above 3.x until recently because of an issue that affected Identity especially (I don't remember the exact details though). That makes me think there might be a decent proportion of users who still use older versions.